File: entities/prescription.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Prescription struct {
	PrescriptionID   uuid.UUID `gorm:"primaryKey" json:"prescription_id"`
	DoctorID         uuid.UUID `gorm:"not null;index" json:"doctor_id"`
	PatientID        string    `gorm:"size:100;not null" json:"patient_id"`
	Diagnosis        string    `gorm:"type:text;not null" json:"diagnosis"`
	MedicinesJSON    string    `gorm:"type:text;not null" json:"medicines"`
	DigitalSignature string    `gorm:"type:text;not null" json:"digital_signature"`
	QRPayload        string    `gorm:"type:text;not null" json:"qr_payload"`
	IssuedAt         time.Time `gorm:"autoCreateTime" json:"issued_at"`
}

func (p *Prescription) BeforeCreate(tx *gorm.DB) (err error) {
	if p.PrescriptionID == uuid.Nil {
		p.PrescriptionID = uuid.New()
	}
	return nil
}

File: backend_api/entities/prescription.py
import uuid
from sqlalchemy import Column, String, DateTime, Text, ForeignKey
from sqlalchemy.sql import func
from database import Base

class Prescription(Base):
    __tablename__ = "prescriptions"

    prescription_id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    patient_id = Column(String(100), nullable=False)
    diagnosis = Column(Text, nullable=False)
    medicines_json = Column(Text, nullable=False)
    digital_signature = Column(Text, nullable=False)
    qr_payload = Column(Text, nullable=False)
    issued_at = Column(DateTime(timezone=True), server_default=func.now())
